components/bl-pdfViewer-component/src/subcomponents.js

```diff
@@ -21,7 +23,31 @@ export function NoData() {
 }
 
 export function Controls(props) {
-  const { pageIndex, setPageIndex, inputRef, handlerPageChange, numPages, display, controlsRef } = props;
+  const { pageIndex, setPageIndex, inputRef, pageCount, display, controlsRef } = props;
+
+  const onSubmit = ({ target }) => {
+    if (Number(target.value)) {
+      const page = ensureRange(target.value, { min: 1, max: pageCount });
+
+      setPageIndex(page);
+      target.value = page;
+    } else {
+      setPageIndex(1);
+      target.value = 1;
+    }
```

Comment:
Also, I not sure, but why we reset it to 1? Maybe will be better to stay at the same page what will be earlier?

For example, we at 5 page, document has 10 pages. And we try to set page 99. Now it will go to 1 page, but better to stay at 5, and make warning/error in console. Or even better, highlith the input and make tooltip that says about non existed page and real range of pages.

`Non existed page "{pageThatWeTryToApply}". Please, choose the page in range {minPage} - {maxPage}`

Reply:
we reset to 1 page only if the value entered is less than 1, or if it is not a number but a string. If we enter a value greater than possible, the last page will open

Reply:
and I think it is better to further limit the input of non-numeric values